src/lib/swapper/swappers/ZrxSwapper/endpoints.ts

```diff
@@ -0,0 +1,173 @@
+      return { id, receiveAddress, affiliateBps, ...tradeQuote }
+    })
+  },
+
+  getUnsignedTx: async ({ from, tradeQuote, stepIndex }: GetUnsignedTxArgs): Promise<ETHSignTx> => {
+    const { accountNumber, buyAsset, sellAsset, sellAmountBeforeFeesCryptoBaseUnit } =
+      tradeQuote.steps[stepIndex]
+
+    const { receiveAddress, recommendedSlippage, affiliateBps } = tradeQuote
+
+    const chainId = sellAsset.chainId
+    const adapterManager = getChainAdapterManager()
+    const adapter = adapterManager.get(chainId) as ChainAdapter<EvmChainId>
+
+    if (adapter === undefined) {
+      throw new SwapError('[executeTrade] - getChainAdapterManager returned undefined', {
+        code: SwapErrorType.UNSUPPORTED_CHAIN,
+        details: { chainId },
+      })
+    }
+
+    if (!isEvmChainAdapter(adapter)) {
+      throw new SwapError('[executeTrade] - non-EVM chain adapter detected', {
+        code: SwapErrorType.EXECUTE_TRADE_FAILED,
+        details: {
+          chainAdapterName: adapter.getDisplayName(),
+          chainId: adapter.getChainId(),
+        },
+      })
+    }
+
+    const maybeZrxQuote = await fetchZrxQuote({
+      buyAsset,
+      sellAsset,
+      receiveAddress,
+      slippage: recommendedSlippage, // TODO: use the slippage from user input
+      affiliateBps,
+      sellAmountBeforeFeesCryptoBaseUnit,
+    })
+
+    if (maybeZrxQuote.isErr()) throw maybeZrxQuote.unwrapErr()
+    const { data: zrxQuote } = maybeZrxQuote.unwrap()
+
+    const { value, to, gasPrice, gas, data } = zrxQuote
+
+    const buildSendApiTxInput = {
+      value: value.toString(),
+      to,
+      from: from!,
+      chainSpecific: {
+        gasPrice: gasPrice.toString(),
+        gasLimit: gas.toString(),
+        maxFeePerGas: undefined,
+        maxPriorityFeePerGas: undefined,
+        data: data.toString(),
+      },
+      accountNumber,
+    }
+    return adapter.buildSendApiTransaction(buildSendApiTxInput)
+  },
+
+  checkTradeStatus: async ({
+    tradeId,
+    txId,
+  }): Promise<{ status: TxStatus; buyTxId: string | undefined; message: string | undefined }> => {
+    try {
+      const maybeTradeQuoteMetadata = tradeQuoteMetadata.get(tradeId)
+      if (!maybeTradeQuoteMetadata) {
+        return createDefaultStatusResponse(txId)
+      }
+      const baseUrl = getUnchainedBaseUrl(maybeTradeQuoteMetadata.chainId)
+      const {
+        data: { status: rawStatus },
+      } = await axios.get<Tx>(`${baseUrl}/api/v1/tx/${txId}`)
+
+      // TODO: write a helper to map all 256 statuses to a status enum + message
+      // https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1066.md#implementation
+      const status = (() => {
+        switch (rawStatus) {
+          case 0:
+            return TxStatus.Failed
+          case 1:
+            return TxStatus.Confirmed
+          default:
+            return TxStatus.Pending
+        }
+      })()
```

Comment:
WRONG
`Broadly applicable status codes for smart contracts.`

we're processing a tx not a contract.

See https://github.com/shapeshift/web/blob/7fe6a2d79b6b6910bf46cacf6d54f3b6c5c6da48/packages/unchained-client/src/evm/parser/index.ts I should create a parser and parse the TX to get the status so we align with rest of app

Reply:
actually i should be using the unchained client in the chain adapter, that way i can get the adapter by chain ID and its already configured with the url